### PR TITLE
Add `badge` config option.

### PR DESCRIPTION
**Pull request by 59naga**
This PR is add the flat-square-badge Inspired by [shields.io](http://shields.io/).

### CODE REVIEW

**Comment by h13i32maru** [comment]
@59naga Hi. Thanks for this PR! But sorry, I do not plan to support this. Please use [ESDoc Plugin API (onComplete)](https://esdoc.org/api.html)


```javascript
exports.onComplete = function(ev) {
  // rewrite the badge xml for you project
};
```

File: manual/configuration.md [hunk 1]
```diff
@@ -110,7 +110,8 @@ npm install esdoc-es7-plugin
     "faq": "./manual/faq.md",
     "changelog": "./CHANGELOG.md"
   },
-  "lint": true
+  "lint": true,
+  "badge": "flat-squre"
```
Comment (StephanBijzitter) [inline manual/configuration.md]: Typo

File: src/Publisher/Builder/CoverageBuilder.js [hunk 2]
```diff
@@ -50,7 +50,7 @@ export default class CoverageBuilder extends DocBuilder {
     } else {
       color = '#4fc921';
     }
-    let badge = this._readTemplate('image/badge.svg');
+    let badge = this._readTemplate(this._config.badge === 'flat-square' ? 'image/badge-flat-square.svg' : 'image/badge.svg');
```
Comment (StephanBijzitter) [inline src/Publisher/Builder/CoverageBuilder.js]: Should use a variable, this is not scalable.

Comment (59naga) [inline src/Publisher/Builder/CoverageBuilder.js]: surely. it was fixed.

**Comment by pixelass** [comment]
The docs are now located here: https://esdoc.org/manual/advanced/api.html
sadly there is no example so it is impossible to understand how this works without digging into the source code.

**Pull request closed without merging**